benchmarks/inference-server/vllm/manifest-templates/vllm.tftpl

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: vllm
+  namespace: ${namespace}
+  labels:
+    app: vllm
+spec:
+  selector:
+    matchLabels:
+      app: vllm
+  template:
+    metadata:
+      labels:
+        app: vllm
+    spec:
+      serviceAccountName: ${ksa}
+      volumes:
+        - name: dshm
+          emptyDir:
+            medium: Memory
+            sizeLimit: 1Gi
+%{ for hugging_face_token_secret in hugging_face_token_secret_list ~}
+        - name: hftoken
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "gcp-secret-provider"
+%{ endfor ~}
+        - name: data
+          hostPath:
+            path: /mnt/stateful_partition/kube-ephemeral-ssd/data
+      containers:
+        - name: vllm
+          ports:
+            - containerPort: 80
```

Comment:
Does vllm have metrics? Can we also add in pod monitoring here to scrape them?

Example from tgi: https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/benchmarks/inference-server/text-generation-inference/monitoring-templates/tgi-podmonitoring.yaml.tftpl

Reply:
Yes. We will add this and HPA config in a follow up change.